url9582d546a9_twentylemon_sudoku_generator_tgz-pJ8-sudoku_Top95J8/src/sudoku/generate/SlowGenerator.java

```diff
@@ -0,0 +1,146 @@
+package sudoku.generate;
+
+import java.util.PriorityQueue;
+import java.util.Random;
+import sudoku.CandidateSet;
+import sudoku.SolverService;
+import sudoku.SudokuBoard;
+import sudoku.SudokuSolver;
+import sudoku.util.ArrayUtil;
+
+/**
+ * Tries to make very few clue puzzles. Only adds the next clue if the
+ * total number of known constraints is maximized among all possibilities.
+ * More specifically, the sum of the cardinalities of all other option sets
+ * should be maximized per each clue addition.
+ *
+ * @author Taras Mychaskiw
+ */
+public class SlowGenerator implements SudokuGenerator {
+
+    private final int[] cells;
+    private final SudokuBoard brd;
+    private final Random rand;
+
+    /**
+     * Constructs a sudoku board generator able to generate boards with the
+     * given dimensions.
+     *
+     * @param p the small box width
+     * @param q the small box height
+     */
+    public SlowGenerator(int p, int q){
+        this.brd = new SudokuBoard(p, q);
+        cells = ArrayUtil.range(0, brd.getNumCells());
+        rand = new Random();
+    }
+
+    /**
+     * Constructs a sudoku board generator able to generate boards with the
+     * same dimensions as the given board.
+     *
+     * @param board a sudoku board
+     */
+    public SlowGenerator(SudokuBoard board){
+        this(board.getSmallWidth(), board.getSmallHeight());
+    }
+
+    /**
+     * Returns a random minimal well formed sudoku with the same dimensions
+     * as specified in creation of this generator.
+     *
+     * @return a random minimal well formed sudoku problem
+     */
+    @Override
+    public SudokuBoard getProblem(){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 101-110 into a separate method.